### PR TITLE
bibcheck: new plugin - delete_subfield_filter.py

### PR DESCRIPTION
**Pull request by ksachs**
* deletes subfield given as argument
* filter on subfield optional

Signed-off-by: Kirsten Sachs <kirsten.sachs@desy.de>

### CODE REVIEW

**Comment by ksachs** [comment]
I can't test it together with rules.cfg.
If possible merge only the bibcheck plugin, so I can test how the rule should look like.

File: bibcheck/rules.cfg [hunk 1]
```diff
@@ -80,3 +80,10 @@ check.field = "999C5s"
 check = add_alice_inspireid
 filter_collection = HEP
 filter_pattern = find cn alice
+
+[rm_isbn_from_jacow_refs]
+check = delete_subfield_filter.py
```
Comment (ksachs) [inline bibcheck/rules.cfg]: should be delete_subfield_filter
without the .py

**Comment by tsgit** [comment]
This leaves an empty MARC tag when there is only the specified subfield

Also, instead of wholesale deletion of a subfield, it might make sense to check the subfield against a pattern. In this case ISBN is known to have a given range of valid lengths

**Comment by ksachs** [comment]
you are right:
bibrecord.record_delete_field   takes care of empty tags.
AmendableRecord.delete_field   does not. Maybe it should.

In this specific case it's no problem - it's never the only subfield.
But maybe someone should update AmendableRecord.delete_field


999C5i is only for isbns. If it is not an ISBN even better to delete it. 
Besides refextract will only put stuff there that looks like a ISBN.

**Pull request merged**